### PR TITLE
correction

### PR DESCRIPTION
**Pull request by egolol**

### CODE REVIEW

**Review by annacanal — APPROVED** [review]

File: braph2/analysis/pvalue2.m [hunk 1]
```diff
@@ -56,11 +56,15 @@
...
                     current_observed_difference = observed_difference(i, j, t);
                     current_random_differences = cellfun(@(x) x(i, j, t), random_differences);
-
-                    P2(i, j, t) =  ...
-                        (length(find(abs(current_random_differences) > abs(current_observed_difference))) + 1) ...
-                        / ...
-                        (length(current_random_differences) + 1); %#ok<AGROW>
+                    
+                    if current_observed_difference == 0 && length(current_random_differences) > 0  %#ok<ISMT>
+                        P2(i, j, t) = 0.05; %#ok<AGROW>
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue2.m:61]: this should be 1

File: braph2/analysis/pvalue1.m [hunk 2]
```diff
@@ -60,14 +60,16 @@
...
                     if current_observed_difference > 0
                         P1(i, j, t) =  ...
-                            (length(find(current_random_differences > current_observed_difference)) + 1) ...
+                            (length(find(current_random_differences > current_observed_difference))) ...
                             / ...
-                            (length(current_random_differences) + 1); %#ok<AGROW>
-                    else  % current_observed_difference <= 0
+                            (length(current_random_differences)); %#ok<AGROW>
+                    elseif current_observed_difference == 0  && length(current_random_differences) > 0 %#ok<ISMT>
+                        P1(i, j, t) =  0.05;                             %#ok<AGROW>
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue1.m:67]: this should be 1

File: braph2/analysis/pvalue2.m [hunk 3]
```diff
@@ -56,11 +56,15 @@
                 for t = 1:1:d_number
                     current_observed_difference = observed_difference(i, j, t);
                     current_random_differences = cellfun(@(x) x(i, j, t), random_differences);
-
-                    P2(i, j, t) =  ...
-                        (length(find(abs(current_random_differences) > abs(current_observed_difference))) + 1) ...
-                        / ...
-                        (length(current_random_differences) + 1); %#ok<AGROW>
+                    
+                    if current_observed_difference == 0 && length(current_random_differences) > 0  %#ok<ISMT>
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue2.m:60]: why do you need the check "&& length(current_random_differences) > 0"?

File: braph2/analysis/pvalue1.m [hunk 4]
```diff
@@ -60,14 +60,16 @@
 
                     if current_observed_difference > 0
                         P1(i, j, t) =  ...
-                            (length(find(current_random_differences > current_observed_difference)) + 1) ...
+                            (length(find(current_random_differences > current_observed_difference))) ...
                             / ...
-                            (length(current_random_differences) + 1); %#ok<AGROW>
-                    else  % current_observed_difference <= 0
+                            (length(current_random_differences)); %#ok<AGROW>
+                    elseif current_observed_difference == 0  && length(current_random_differences) > 0 %#ok<ISMT>
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue1.m:66]: why do you need the check "&& length(current_random_differences) > 0"?

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
The default p-value should be 1, not 0.05, because we don't what it significant if the differences are zero.

I think the second check should not be "length(current_random_differences) > 0", but something to double-check that all values in current_random_differences are zero.

**Comment by giovannivolpe** [comment]
@egolol @annacanal @mitemijalkov Also, please add some unit tests to check that when the two distributions are the same the pvalues are not significative.

File: braph2/analysis/pvalue1.m [hunk 5]
```diff
@@ -60,14 +60,16 @@
...
+                        P1(i, j, t) =  0.05;                             %#ok<AGROW>
+                    else  % current_observed_difference < 0
                         P1(i, j, t) =  ...
-                            (length(find(current_random_differences < current_observed_difference)) + 1) ...
+                            (length(find(current_random_differences < current_observed_difference))) ...
                             / ...
-                            (length(current_random_differences) + 1); %#ok<AGROW>
+                            (length(current_random_differences)); %#ok<AGROW>
                     end
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue1.m:73]: add if pvalue == 0, make it 1/number premutations

File: braph2/analysis/pvalue2.m [hunk 6]
```diff
@@ -56,11 +56,15 @@
...
+                        P2(i, j, t) = 0.05; %#ok<AGROW>
+                    else
+                        P2(i, j, t) =  ...
+                            (length(find(abs(current_random_differences) > abs(current_observed_difference)))) ...
+                            / ...
+                            (length(current_random_differences)); %#ok<AGROW>
+                    end
                 end
```
Comment (giovannivolpe) [inline braph2/analysis/pvalue2.m:68]: add if pvalue == 0, make it 1/number premutations

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

File: braph2/analysis/pvalue1.m (hunk 5 above)
Comment (egolol) [reply braph2/analysis/pvalue1.m:73]: I dont understand this instruction. Is it : 

1.  if P1(i, j, t) = 0   then P1(i,j,t) = 1/ # permutations end

2. P1(i,j,t) = (length(find(current_random_differences < current_observed_difference))) ...
                            / ...
                            length(random_differences);

Comment (annacanal) [reply braph2/analysis/pvalue1.m:73]: Yes, I understand it like this. well, 2 before 1.

**Pull request closed without merging**